backend/src/youtube-feed/youtube-feed.service.ts

```diff
@@ -30,20 +38,32 @@ export class YouTubeFeedService {
     }
 
     onModuleInit() {
-        this.fetchAndSaveYouTubeFeed();
+        setInterval(() => {
+            this.fetchAndSaveYouTubeFeed();
+        }, 1000);
     }
 
     async fetchAndSaveYouTubeFeed() {
         this.logger.log('Fetching Youtube feeds Started');
         try {
-            const feed = await axios.get(youtube_fetch_url);
-            const videoIds = feed.data.items.map(item => item.id.videoId);
-            const videoDetails = await axios.get(youtube_video_details_url + videoIds.filter(id => id));
-            const youTubeFeeds = feed.data.items.map(item => {
+            const channelDetails = await axios.get(channelDetailsUrl);
+
+            const channelTitle = channelDetails.data.items[0].snippet.title;
+            const channelProfileImage = channelDetails.data.items[0].snippet.thumbnails.default.url;
+
+            const feed = await axios.get(youtubeFetchUrl);
+            let youTubeFeeds = feed.data.items.filter(item => item.id.videoId)
+            youTubeFeeds = await this.filterAlreadyExistingFeeds(youTubeFeeds);
+
+            const videoIds = youTubeFeeds.map(item => item.id.videoId);
+            const videoDetails = await axios.get(youtubeVideoDetailsUrl + videoIds);
+
+            youTubeFeeds = youTubeFeeds.map(item => {
                 const details = videoDetails.data.items.find(videoItem => videoItem.id === item.id.videoId);
                 return {
+                    author: channelTitle,
```

Comment:
Looks like transformational code.
Pull this out to a function called `formatData` instead?